sw/host/opentitanlib/src/util/present.rs

```diff
@@ -0,0 +1,314 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+
+/// PRESENT block cipher.
+///
+/// Based on version 1.2 of the following Python implementation
+/// https://github.com/doegox/python-cryptoplus
+pub struct Present {
+    round_keys: Vec<u64>,
+    rounds: usize,
+}
+
+impl Present {
+    pub fn try_new_rounds(key: &[u8], rounds: usize) -> Result<Present> {
```

Comment:
Would it make more sense to accept a `Vec` by value, since you're going to copy it anyways?